File: benchmarks/primes/primes.c
#include <stdio.h>

int is_prime(int n) {
    if (n < 2) return 0;
    if (n < 4) return 1;
    if (n % 2 == 0 || n % 3 == 0) return 0;
    int i = 5;
    while (i * i <= n) {
        if (n % i == 0 || n % (i + 2) == 0) return 0;
        i += 6;
    }
    return 1;
}

int main() {
    int count = 0;
    for (int n = 2; n <= 120000000; n++) {
        if (is_prime(n)) count++;
    }
    printf("%d\n", count);
    return 0;
}

File: benchmarks/primes/primes.go
package main

import "fmt"

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n < 4 {
		return true
	}
	if n%2 == 0 || n%3 == 0 {
		return false
	}
	for i := 5; i*i <= n; i += 6 {
		if n%i == 0 || n%(i+2) == 0 {
			return false
		}
	}
	return true
}

func main() {
	count := 0
	for n := 2; n <= 120000000; n++ {
		if isPrime(n) {
			count++
		}
	}
	fmt.Println(count)
}
